### ISSUE TITLE
Regression after latest update

### ISSUE THREAD

**Issue by rainerborene** [palkan/action_policy#123]
Our app broke after upgrading to the latest version. For some reason the following statement is returning the wrong value now. It's returning `CoursePolicy`, but should be `nil` as before, correct?
```ruby
ActionPolicy.lookup(:courses, namespace: Admin, allow_nil: true) # returning CoursePolicy (wrong)
```

**Comment by palkan** [comment]
I believe this relates to https://github.com/palkan/action_policy/issues/122

There was a bug with namespaces support for symbols: we didn't take into account modules nesting, thus the behavior differs from the baseline (record/model-based, the one that is described in [the docs](https://actionpolicy.evilmartians.io/#/namespaces)).
For example, `ActionPolicy.lookup(Course.new, namespace: Admin, allow_nil: true)` always returned `CoursePolicy`, that's the expected behaviour. And not it's fixed for symbols.

Your use-case testifies that we need something like `strict_namespace` option or whatever to disable the current fallback mechanism.

**Comment by rainerborene** [comment]
I've been using this bug as a feature so long then :rofl:. But I think you're right: `strict_namespace` would be the obvious solution.

**Comment by palkan** [comment]
And I guess we need a `fallback` option, too: `ActionPolicy.lookup(target, namespace: Admin, fallback: Admin::DefaultPolicy, namespace_nesting: false)`

**Comment by palkan** [comment]
Closed by #124. Will be released in 0.5.0.
